app/controllers/carts_controller.rb

```diff
@@ -6,7 +6,11 @@ def create
     @cart.add_trip(trip.id)
     session[:cart] = @cart.contents
     flash[:success] = "You have #{pluralize(@cart.count_of(trip.id), "#{trip.title} trip")} in your cart."
-    redirect_to trips_path
+    if request.env["HTTP_REFERER"].include?("/trips/")
```

Comment:
This is interesting. We should keep this in mind for potential weird errors in the future.